packages/app-desktop/gui/NoteEditor/utils/useFormNote.ts

```diff
@@ -189,13 +189,23 @@ export default function useFormNote(dependencies: HookDependencies) {
 
 			const focusSettingName = noteIsTodo ? 'newTodoFocus' : 'newNoteFocus';
 
-			requestAnimationFrame(() => {
-				if (Setting.value(focusSettingName) === 'title') {
-					if (titleInputRef.current) titleInputRef.current.focus();
-				} else {
-					if (editorRef.current) editorRef.current.execCommand({ name: 'editor.focus' });
+			function autoFocusLoop() {
+				if (!editorRef.current) { // Wait for editorRef to load
+					setTimeout(autoFocusLoop, 100);
+					return;
 				}
-			});
+
+				requestAnimationFrame(() => {
+					if (Setting.value(focusSettingName) === 'title') {
+						if (titleInputRef.current) titleInputRef.current.focus();
+					} else {
+						if (editorRef.current) {
+							editorRef.current.execCommand({ name: 'editor.focus' });
+						}
+					}
+				});
+			}
+			autoFocusLoop();
```

Comment:
This is much better, but as you see you're within an effect here and that effect can be cleared at any time. When that will happen your timer will be not cleared and run indefinitely. So have a look at the structure of the effect and see how you can clear it

Reply:
Apologies, I knew I was missing something but couldn't quite figure out what it was at the time, so thank you for pointing that out to me. My first thought was to set the timeout to a variable and use clearTimeout in the cleanup function but that didn't work properly, so I sort of borrowed your approach and created an auto focus cancelled flag that gets set to true in the cleanup function.